pandas/core/construction.py

```diff
@@ -402,6 +402,24 @@ def extract_array(obj: object, extract_numpy: bool = False) -> Union[Any, ArrayL
     return obj
 
 
+def wrap_if_datetimelike(arr):
```

Comment:
can you type the arg (and return dtype)

Reply:
not really, its unrestricted

Reply:
aren't these *always* TDI/DTI on the return type? (sure input is just ArrayLike)

Reply:
no, we call it on arbitrary inputs in ops.array_ops

Reply:
ok then i am misunderstanding how this is used and its definetly maybe_wrap_if_datetimelike, its either ensure or maybe, really shouldn't be any other choices.

Reply:
ok will update

Reply:
renamed + greenish